source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -71,58 +117,258 @@ absl::optional<Arn> Filter::calculateRouteArn() {
     return absl::nullopt;
   }
 
-  return parseArn(settings->arn());
+  return *settings;
+}
+
+std::string Filter::resolveSettings() {
+  if (auto route_settings = getRouteSpecificSettings()) {
+    if (auto route_arn = parseArn(route_settings->arn())) {
+      arn_.swap(route_arn);
+      payload_passthrough_ = route_settings->payloadPassthrough();
+    } else {
+      // TODO(marcomagdy): add stats for this error
+      ENVOY_LOG(debug, "Found route specific configuration but failed to parse Lambda ARN {}.",
+                route_settings->arn());
+      return "Invalid AWS Lambda ARN";
+    }
+  } else {
+    payload_passthrough_ = settings_.payloadPassthrough();
+  }
+  return {};
 }
 
 Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers, bool end_stream) {
-  if (!settings_.payloadPassthrough()) {
+  auto cluster_info_ptr = decoder_callbacks_->clusterInfo();
+  ASSERT(cluster_info_ptr);
```

Comment:
I don't think you can guarantee that this is non-null so you should handle this case. See other examples.